packages/asset-swapper/src/utils/market_operation_utils/fill_optimizer.ts

```diff
@@ -32,24 +32,27 @@ export class FillsOptimizer {
     public optimize(fills: Fill[], input: BigNumber, upperBoundPath?: Fill[]): Fill[] | undefined {
         this._currentRunCount = 0;
         this._optimalPath = upperBoundPath;
-        this._optimalPathOutput = upperBoundPath ? getPathOutput(upperBoundPath, input) : ZERO_AMOUNT;
+        this._optimalPathAdjustedOutput = upperBoundPath ? getPathAdjustedOutput(upperBoundPath, input) : ZERO_AMOUNT;
         const ctx = {
             currentPath: [],
             currentPathInput: ZERO_AMOUNT,
-            currentPathOutput: ZERO_AMOUNT,
+            currentPathAdjustedOutput: ZERO_AMOUNT,
             currentPathFlags: 0,
         };
         // Visit all valid combinations of fills to find the optimal path.
         this._walk(fills, input, ctx);
-        return this._optimalPath;
+        if (this._optimalPath) {
+            return sortFillsByAdjustedRate(this._optimalPath, this._shouldMinimize);
```

Comment:
This can return ALL native orders if the `runLimit` is reached. Problem with that is the amount of liquidity available can far out weigh the input liquidity. I.e a 10 USDC -> DAI sell can return 20+ 0x orders. This has some downstream issues with protocol fee calculation and gas costs.

Reply:
Added order pruning (but not clipping) to ameliorate this issue.